tizen-api/src/tizen-api-pipeline.c

```diff
@@ -283,18 +308,57 @@ nns_pipeline_construct (const char *pipeline_description, nns_pipeline_h * pipe)
 int
 nns_pipeline_destroy (nns_pipeline_h pipe)
 {
-  /* nns_pipeline *p = pipe; */
+  nns_pipeline *p = pipe;
+  GstStateChangeReturn scret;
+  GstState state, pending;
+
+  if (p == NULL)
+    return NNS_ERROR_INVALID_PARAMETER;
+
+  g_mutex_lock (&p->lock);
+
+  /* Check if it's PLAYING */
+  scret = gst_element_get_state (p->element, &state, &pending, 10000000UL);     /* 10ms */
+  if (scret == GST_STATE_CHANGE_SUCCESS && state == GST_STATE_PLAYING) {
```

Comment:
update to `if ((scret == GST_STATE_CHANGE_SUCCESS || scret == GST_STATE_CHANGE_ASYNC) && state == GST_STATE_PLAYING)`
This adds the case, with live source & sink - live source is in playing state, and is generating/generated first data buffer. However, the first data buffer has not reached the sink yet, and sink/bin/pipeline returns `GST_STATE_CHANGE_ASYNC`.

Reply:
As noted above, I'll ignore get-state status and force it stop & destroyed.